src/player.cpp

```diff
@@ -4599,6 +4599,73 @@ void player::on_hurt( Creature *source, bool disturb /*= true*/ )
     }
 }
 
+bool player::immune_to( body_part bp, damage_unit dam ) const
+{
+    // We use the worst-case scenario instead of random numbers
+    if( has_trait( "DEBUG_NODMG" ) || is_immune_damage( dam.type ) ) {
+        return true;
+    }
+    if( dam.type == DT_HEAT ) {
+        return false; // No one is immune to fire
```

Comment:
Not really true, especially with the `is_immune_damage` check above.

For purposes of this function, it would be better if the `is_immune_damage` check was below this check for fire damage. It is possible to be immune to fire, but the clothing will still burn, meaning the warnings for entering fire should still happen.